root_VS2012/programs/C#/Samples/WebApp_sample/MVC_Sample/MVC_Sample/Controllers/CrudMu2Controller.cs

```diff
@@ -654,5 +654,21 @@ public ActionResult Transitions()
         {
             return Redirect(Url.Action("Index", "CrudMu"));
         }
+
+        /// <summary>
+        /// Sleepを実行し二重送信防止機能をテストする
+        /// </summary>        
+        /// <returns>再描画（ViewResult）</returns>
+        public ActionResult PreventDoubleSubmission(FormCollection form)
+        {
+            System.Threading.Thread.Sleep(5 * 1000);
+
+            CrudModel model = new CrudModel();
+            // 値の復元のため、CopyInputValuesを呼び出す。
```

Comment:
@yusukemaegawa 

> I think that these should not be split.

Because it become parts,
I think that it is not good to explain a series of all processing.